addons/mail/models/mail_message.py

```diff
@@ -848,12 +857,145 @@ def _notify(self, force_send=False, send_after_commit=True, user_signature=True)
+                    ])
+
+        channel_ids = messages_to_delete.mapped('res_id')
+        moderators = messages_to_delete.env['mail.channel'].browse(channel_ids).mapped('moderator_ids')
+        authors = messages_to_delete.mapped('author_id')
+
+        moderators_notifications = [
+            (
+                moderator.partner_id.id,
+                [
+                    message.id
+                    for message in messages_to_delete
+                    if message.res_id in moderator.moderated_channel_ids.ids
+                ]
+            )
+            for moderator in moderators
+        ]
+
+        authors_notifications = [
+            (
+                author.id,
+                [
+                    message.id
+                    for message in messages_to_delete
+                    if message.author_id == author
+                ]
+            )
+            for author in authors
+        ]
+
+        brut_notifications = moderators_notifications + authors_notifications
+
+        # The following steps ensures that no message id is sent twice to the same person
+        # (in case the person is both moderator and author of the message).
+        dictionary = {}
+        for notif in brut_notifications:
+            partner_id = notif[0]
+            message_ids = notif[1]
+            if partner_id in dictionary:
+                dictionary[partner_id] = list(set(dictionary[partner_id]) or set(message_ids))
+            else:
+                dictionary[partner_id] = message_ids
+
+        nice_notifications = []
+        for partner_id, message_ids in dictionary.items():
+            nice_notifications.append(
+                        [
+                            (messages_to_delete._cr.dbname, 'res.partner', partner_id),
+                            {
+                                'type': 'deletion',
+                                'message_ids': dictionary[partner_id]
+                            }
+                        ]
+                    )
+        messages_to_delete.env['bus.bus'].sendmany(nice_notifications)
+        messages_to_delete.unlink()
+
+    @api.model
+    def _notify_moderators(self):
+        """ This method alerts by email the moderators of the existence of messages that need moderation.
+            This method is called once a day by a cron.
+            """
+        moderators_to_notify = self.search([('moderation_status', '=', 'pending_moderation')]).mapped('channel_ids.moderator_ids')
+        template = self.env.ref('mail.mail_template_moderator_notification_email', raise_if_not_found=False)
+        for moderator in moderators_to_notify:
+            template.with_context(lang=moderator.lang).send_mail(moderator.partner_id.id, raise_exception=True)
+
+    @api.model
+    def create_notification_email(self, vals):
+        self.env['mail.mail'].create(vals).mark_outgoing()
+
+    def _moderator_message_notifications(self, message):
```

Comment:
I would improve the naming (this remark is valable for other methods, I did not always note it) : like _notify_moderators .

Reply:
done